pkg/config/read.go

```diff
@@ -158,11 +165,11 @@ func readCmdline() (map[string]interface{}, error) {
 	}
 
 	data := map[string]interface{}{}
-	for _, item := range strings.Fields(string(bytes)) {
+	for _, item := range parser.FindAllString(string(bytes), -1) {
 		parts := strings.SplitN(item, "=", 2)
 		value := "true"
 		if len(parts) > 1 {
-			value = parts[1]
+			value = strings.Trim(parts[1], `"`)
```

Comment:
For reasons described at https://github.com/rancher/k3os/pull/218#issuecomment-541870309
you will want to `strings.TrimLeft` `parts[0]` and `strings.TrimRight` `parts[1]` with the same custset as used here.

Reply:
So I tested booting via Qemu straight from initrd+kernel and I am pretty sure grub is doing the munging. This means we have to operate correctly under both conditions, e.g.
1. `boot_cmd="echo TEST"`
2. `"boot_cmd=echo TEST"`

Please update the regex to handle both scenarios and then augment your trimming accordingly.